crates/lang/ir/src/ir/attrs.rs

```diff
@@ -230,6 +230,12 @@ impl InkAttribute {
         })
     }
 
+    /// Returns `true` if the ink! attribute contains the `constructor` argument.
+    pub fn is_constructor(&self) -> bool {
+        self.args()
+            .any(|arg| matches!(arg.kind(), AttributeArgKind::Constructor))
```

Comment:
While this works a message is only a constructor if the first argument is #[ink(constructor)]`. This is to disambiguate messages and constructors that have been invalidly flagged with both.